src/lldp_syncd/daemon.py

```diff
@@ -367,18 +387,32 @@ def sync(self, parsed_update):
                 logger.debug("sync'd: {}".format(json.dumps(chassis_update, indent=3)))
 
         new, changed, deleted = self.cache_diff(self.interfaces_cache, parsed_update)
+
+        # For changed elements, if only lldp_rem_time_mark changed, update its value, otherwise delete and repopulate
+        for interface in changed:
+            if re.match(SONIC_ETHERNET_RE_PATTERN, interface) is None:
+                logger.warning("Ignoring interface '{}'".format(interface))
+                continue
+            table_key = ':'.join([LldpSyncDaemon.LLDP_ENTRY_TABLE, interface])
+            if self.is_only_time_mark_modified(self.interfaces_cache[interface], parsed_update[interface]):
+                self.db_connector.set(self.db_connector.APPL_DB, table_key, 'lldp_rem_time_mark', parsed_update[interface]['lldp_rem_time_mark'], blocking=True)
+                logger.debug("Only sync'd interface {} lldp_rem_time_mark: {}".format(interface, parsed_update[interface]['lldp_rem_time_mark']))
+            else:
+                self.db_connector.delete(self.db_connector.APPL_DB, table_key)
```

Comment:
Why delete and hmset? As my understand hmset will always add the data back.

Reply:
Actually following code can improve:

            if self.is_only_time_mark_modified(self.interfaces_cache[interface], parsed_update[interface]):
                self.db_connector.set(self.db_connector.APPL_DB, table_key, 'lldp_rem_time_mark', parsed_update[interface]['lldp_rem_time_mark'], blocking=True)
                logger.debug("Only sync'd interface {} lldp_rem_time_mark: {}".format(interface, parsed_update[interface]['lldp_rem_time_mark']))
            else:
                self.db_connector.delete(self.db_connector.APPL_DB, table_key)
                self.db_connector.hmset(self.db_connector.APPL_DB, table_key, parsed_update[interface])
                logger.debug("Sync'd changed interface {} : {}".format(interface, parsed_update[interface]))



Because if only lldp_rem_time_mark changed, then hmset result will exactly same with what you do by hset.

Maybe the code can be improve to this?

            if parsed_update[interface].length == 0: <== delete when interface data is empty?
                self.db_connector.delete(self.db_connector.APPL_DB, table_key)
            else:
                self.db_connector.hmset(self.db_connector.APPL_DB, table_key, parsed_update[interface])

Reply:
Per the offline discussion, there could be scenario that changed interfaces have less element than before, hmset could just update the changed elements not remove non-existing ones. For example, the interface has 8 elements before, next time, it has only 7, hmset probably still keeps 8 elements in database.
Considering this scenario, we agree to keep the current logic to make sure data is updated accurately.